web/src/components/organisms/Settings/Workspace/index.tsx

```diff
@@ -0,0 +1,198 @@
+import styled from "@emotion/styled";
+import Button from "@reearth-cms/components/atoms/Button";
+import Form from "@reearth-cms/components/atoms/Form";
+import Input from "@reearth-cms/components/atoms/Input";
+import List from "@reearth-cms/components/atoms/List";
+import React, { useCallback, useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+
+import useHooks, { RoleUnion } from "./hooks";
+
+const Workspace: React.FC = () => {
+  const { workspaceId } = useParams();
+  const {
+    me,
+    currentWorkspace,
+    searchedUser,
+    changeSearchedUser,
+    deleteWorkspace,
+    searchUser,
+    updateName,
+    addMembersToWorkspace,
+    updateMemberOfWorkspace,
+    removeMemberFromWorkspace,
+  } = useHooks({ workspaceId });
+  const [owner, setOwner] = useState(false);
+  const [memberName, setMemberName] = useState<string>("");
+  const [workspaceName, setWorkspaceName] = useState<string>("");
+  const navigate = useNavigate();
+  const members = currentWorkspace?.members;
+
+  const handleMemberNameChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setMemberName?.(e.currentTarget.value);
+      searchUser(e.currentTarget.value);
+    },
+    [setMemberName, searchUser]
+  );
+
+  const handleAddMember = useCallback(() => {
+    if (!searchedUser) return;
+    addMembersToWorkspace([searchedUser.id]);
+    setMemberName("");
+    changeSearchedUser(undefined);
+  }, [searchedUser, addMembersToWorkspace, changeSearchedUser]);
+
+  const handleDeleteWorkspace = useCallback(async () => {
+    await deleteWorkspace();
+    navigate("/workspace");
+  }, [deleteWorkspace, navigate]);
+
+  const handleDeleteMember = useCallback(
+    async (userId: string) => {
+      await removeMemberFromWorkspace(userId);
+    },
+    [removeMemberFromWorkspace]
+  );
```

Comment:
Is there a reason you make separate callbacks in the hooks.ts file, and then wrap them again in callbacks here in the component?
It feels redundant and the code for both files are very long, reducing the benefits of a hooks file.